### ISSUE TITLE
Hyper doesn't set the environment variable $WINDOWID

### ISSUE THREAD

**Issue by dylanaraps** [zeit/hyper#2037]
- [x] I am on the [latest](https://github.com/zeit/hyper/releases/latest) Hyper.app version
- [x] I have searched the [issues](https://github.com/zeit/hyper/issues) of this repo and believe that this is not a duplicate

- **OS version and name**: Arch Linux
- **Hyper.app version**: `1.3.3`
- **Link of a [Gist](https://gist.github.com/) with the contents of your .hyper.js**: No changes made to default config. (N/A)
- **Relevant information from devtools**: N/A
- **The issue is reproducible in vanilla Hyper.app**: Yes

## Issue

`hyper` doesn't seem to be setting the environment variable `$WINDOWID`. Various applications on Linux use this to function, namely `w3m-img` (Terminal image viewer) which uses the variable to identify which window to display the images in.

Without this environment variable being set `w3m-img` fails to locate the window to display images in and fails to function entirely.

All of the other terminal emulators I've tried set this variable from minimalist terminal emulators like `st` and `alacritty` to more featureful terminal emulators like `xterm`, `URxvt` and the VTE family (`GNOME-terminal`, `XFCE4-terminal`, `Konsole`, `termite`.)

The point I'm trying to get across is that I haven't found a terminal emulator that **doesn't** set this. 

More information:

- https://github.com/jwilm/alacritty/issues/471
- https://github.com/jwilm/alacritty/pull/586

**Comment by egmontkob** [comment]
> w3m-img (Terminal image viewer) which uses the variable to identify which window to display the images in

Yup that's a terrible hack that breaks in plenty of ways. :) E.g. cannot take the tab bar's height into account.  Or if you drag-n-drop the terminal emulator tab to another window, it'll still pop up over the previous window as there's no way to update the `WINDOWID` environment variable inside the already running shell.

> Various applications on Linux use this to function

w3m-img is the only one I'm aware of. What else is there?

> The point I'm trying to get across is that I haven't found a terminal emulator that doesn't set this.

Wayland (the new protocol eventually aiming to replace X11, already the default in Fedora and Ubuntu) does not allow windows to position themselves, so this can no longer work. Plus, the concept of window ID, if exists at all, is probably incompatible with X11's. `gnome-terminal` doesn't set `WINDOWID` under Wayland.

That being said, of course `hyper` could set it when running on X11, it wouldn't hurt.

**Comment by albinekb** [comment]
Thanks for the info and excellent thoughts @egmontkob 👍  As you said, if there's not too much code (i don't think this needs much) a PR is welcome to set this variable on linux when running under X11.
